src/vmm/src/lib.rs

```diff
@@ -1043,4 +1043,44 @@ mod tests {
             assert!(vmm.add_net_device(&cfg).is_err());
         }
     }
+
+    #[test]
+    #[cfg(target_arch = "aarch64")]
+    fn test_setup_fdt() {
+        let vmm_config = default_vmm_config();
+        let mut vmm = mock_vmm(vmm_config);
+
+        {
+            let result = vmm.setup_fdt();
+            assert!(result.is_ok());
+        }
+
+        {
+            let mem_size: u64 = vmm.guest_memory.iter().map(|region| region.len()).sum();
+            let fdt_offset = mem_size - AARCH64_FDT_MAX_SIZE - 0x10000;
+            let fdt = FdtBuilder::new()
+                .with_cmdline(vmm.kernel_cfg.cmdline.as_str())
+                .with_num_vcpus(vmm.num_vcpus.try_into().unwrap())
+                .with_mem_size(mem_size)
+                .with_serial_console(0x40000000, 0x1000)
+                .with_rtc(0x40001000, 0x1000)
+                .create_fdt()
+                .unwrap();
+            assert!(fdt.write_to_mem(&vmm.guest_memory, fdt_offset).is_ok());
+        }
```

Comment:
Hmm, what is the reason for adding this test case? Looks like it simply copies the code from `setup_fdt`, and you already check above that `setup_fdt` works as expected.

Reply:
Yes, this is a duplicate, I will delete it